Automattic/wp-calypso
# Convert `WebServerSettingsCard` to Typescript

Related to #93452

## Proposed Changes

Convert `WebServerSettingsCard` in `client/my-sites/hosting/web-server-settings-card/index.tsx` to Typescript. The biggest change I had to make was dropping `react-redux`s `connect` function in favor of `useSelector` calls. This saves us from having to manually specify types, which would be very tedious.

## Why are these changes being made?


I was working on this file in #93452 and found it helpful to do some housekeeping in this way.

## Testing Instructions



1. Have an atomic website
2. Apply the following diff

```diff
diff --git a/client/my-sites/hosting/sftp-card/index.js b/client/my-sites/hosting/sftp-card/index.js
index dfe46f0adb..8778ccb120 100644
--- a/client/my-sites/hosting/sftp-card/index.js
+++ b/client/my-sites/hosting/sftp-card/index.js
@@ -308,15 +308,6 @@ export const SftpCard = ( {
                                                <SftpSshLabel htmlFor="ssh">{ translate( 'SSH access' ) }</SftpSshLabel>
                                        ) }
                                        { siteHasSshFeature && renderSshField() }
-                                       <ReauthRequired twoStepAuthorization={ twoStepAuthorization }>
-                                               { () => (
-                                                       <>
-                                                               { siteHasSshFeature && isSshAccessEnabled && (
-                                                                       <SshKeys disabled={ disabled } siteId={ siteId } siteSlug={ siteSlug } />
-                                                               ) }
-                                                       </>
-                                               ) }
-                                       </ReauthRequired>
                                </FormFieldset>
                        ) }
                        { isLoading && <Spinner /> }
```

3. Navigate to `/hosting-config/:site_slug`
4. Ensure that clicking the WP version in the header scrolls the Web server settings card into view and that the `WordPress version` text blinks
5. Ensure the same thing for the PHP version in the header
6. Update the `PHP version` and ensure that the operation completes successfully
7. Update the `Handling requests for nonexistent assets` and ensure that the operation completes successfully
8. Have a staging site
9. Navigate to `/hosting-config/:site_slug` for the staging site
10. Update the `WordPress version` and ensure that the operation completes successfully
